src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonParameterInfo.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Reflection;
+using System.Text.Json.Serialization;
+
+namespace System.Text.Json
+{
+    [DebuggerDisplay("ParameterInfo={ParameterInfo}")]
+    internal abstract class JsonParameterInfo
```

Comment:
Please add a comment about what this class does. Is "JsonConstructorParameterInfo" better? Or could this be used in non-ctor cases in the future?

Reply:
Added.

> Or could this be used in non-ctor cases in the future?

Yes. For this reason I prefer to leave it as "JsonParameterInfo".